primary_beam_correction.py

```diff
@@ -0,0 +1,296 @@
+#!/usr/bin/env python
+import argparse
+import logging
+import numpy as np
+from astropy import units as u
+from astropy.io import fits
+from astropy import wcs
+from astropy import coordinates
+
+
+def read_fits(pathfits):
+    '''
+    This functions reads in the fits file
+    Input: Path to the fits file
+    Output: Fits file object
+    '''
+    fl = fits.open(pathfits)
+    images = fl[0]
+    return images
+
+
+def get_positions(path):
+    '''
+    This function reads the fits file and returns the sky coordinate
+    of the phase centre.
+    Input: path to the fits file
+    Output: Sky coordinate of the phase centre
+    '''
+    images = read_fits(path)
+    # Get pointing centre of observation
+    phase_centre_ra = images.header['OBSRA']
+    phase_centre_dec = images.header['OBSDEC']
+    # Convert to astropy.coordinates.SkyCoord object
+    phase_center = coordinates.SkyCoord(phase_centre_ra, phase_centre_dec, unit=(u.deg, u.deg))
+    # Get positions of other sources
+    source_pos = wcs.WCS(images.header)
+    return phase_center, source_pos
+
+
+def radial_offset(phase_center, source_pos, row, col):
+    '''
+    This function gets the position in ra and dec of a pixel and determine its
+    radial offset from phase centre.
+    Input: phase center and source positions
+    output: A separation vector between phase centre and source positions in radians.
+    '''
+    p2w = source_pos.pixel_to_world(row, col, 0, 0)[0]
```

Comment:
`source_pos` is a misleading variable name. The positions are of image pixels not sources. In astronomy a 'source' normally refers to some kind of emmiting object (e.g a radio source, is a galaxy/ HII region etc.)

Reply:
I have a bad habit when it comes to variable names, I promise to improve. I have dealt with this one.